src/app/clusters/concentration-measurement-server/concentration-measurement-server.h

```diff
@@ -325,256 +372,200 @@ class Instance : public AttributeAccessInterface
         return CHIP_NO_ERROR;
     };
 
+    template <bool Enabled = NumericMeasurementEnabled, typename = std::enable_if_t<Enabled, CHIP_ERROR>>
     CHIP_ERROR SetMeasuredValue(DataModel::Nullable<float> aMeasuredValue)
     {
-        if constexpr (NumericMeasurementEnabled)
+        if (!CheckConstraintMinMax(aMeasuredValue, this->mMinMeasuredValue, this->mMaxMeasuredValue))
         {
-            if (!CheckConstraintMinMax(aMeasuredValue, mMinMeasuredValue, mMaxMeasuredValue))
-            {
-                return CHIP_ERROR_INVALID_ARGUMENT;
-            }
-
-            DataModel::Nullable<float> oldValue = mMeasuredValue;
-            mMeasuredValue                      = aMeasuredValue;
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
 
-            if (oldValue != mMeasuredValue)
-            {
-                ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::MeasuredValue::Id);
-                MatterReportingAttributeChangeCallback(path);
-            }
+        DataModel::Nullable<float> oldValue = this->mMeasuredValue;
+        this->mMeasuredValue                = aMeasuredValue;
 
-            return CHIP_NO_ERROR;
-        }
-        else
+        if (oldValue != this->mMeasuredValue)
         {
-            return CHIP_ERROR_INCORRECT_STATE;
+            ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::MeasuredValue::Id);
+            MatterReportingAttributeChangeCallback(path);
```

Comment:
Not directly related to this PR, but is this smaller or larger code (or the same) compared to doing:
```
            MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::MeasuredValue::Id);
```
?  Worth measuring so we know.

Reply:
840 bytes smaller doing it with your suggestion. I also think thats way more readable, so will push that, thanks.